app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -125,6 +125,19 @@ def resource_class(m=nil)
     mapping.to
   end
 
+  def get_resource(uid_field='uid', uid=nil)
+    if DeviseTokenAuth.custom_scope
+      resource_class.send(DeviseTokenAuth.custom_scope, uid_field, uid).first
```

Comment:
Not so sure that with the custom scope that the `uid_field` is a necessary parameter when using a scope... But I guess it doesn't hurt anything. Here's what it'd look like from the model level:

```ruby
class User < ActiveRecord::Base
  scope :my_custom_scope, ->(uid_field, uid){
    where(channel_id: "WW", email: uid) # Not really any purpose of using the uid_field here IMO
  }
end
```